modules/actor/actor-wfrp4e.js

```diff
@@ -3606,19 +3584,23 @@ export default class ActorWfrp4e extends Actor {
       return ui.notifications.error(game.i18n.localize("ERROR.EffectNotFound"))
 
     effect.origin = this.uuid;
+    let duration
 
     let multiplier = 1
-    if (test && test.result.overcast && test.result.overcast.usage.duration)
-      multiplier += test.result.overcast.usage.duration.count
+    if (test && test.result.overcast && test.result.overcast.usage.duration) {
+      duration = test.result.overcast.usage.duration.current;
+    } else if(item.Duration) {
+      duration = parseInt(item.Duration);
+    }
 
-    if (item.duration && item.duration.value.toLowerCase().includes(game.i18n.localize("minutes")))
-      effect.duration.seconds = parseInt(item.Duration) * 60 * multiplier
+    if (duration && item.duration.value.toLowerCase().includes(game.i18n.localize("Minutes")))
+      effect.duration.seconds = duration * 60 * multiplier
 
-    else if (item.duration && item.duration.value.toLowerCase().includes(game.i18n.localize("hours")))
-      effect.duration.seconds = parseInt(item.Duration) * 60 * 60 * multiplier
+    else if (duration && item.duration.value.toLowerCase().includes(game.i18n.localize("Hours")))
+      effect.duration.seconds = duration * 60 * 60 
 
-    else if (item.duration && item.duration.value.toLowerCase().includes(game.i18n.localize("rounds")))
-      effect.duration.rounds = parseInt(item.Duration) * multiplier
```

Comment:
> whenever an overcast is used for duration - it is not applied for active effect, because test is recreated from data inside message.

I can't seem to reproduce this. The test (and message) data should be updated with every overcast and calculate the correct value in `usage.duration`

Additionally your fix uses the same object, so why would that be different? (`duration.count` vs `duration.current`

Reply:
so, the change here came during my testing and was purely cosmetical. The only potential benefit of this was to use exact duration in case there are any other homebrew in use. Anyway, here is example where problem lies: 

![image](https://user-images.githubusercontent.com/7930626/200282246-322fac1b-35a0-4f35-bacb-9819e3dfd820.png)

The roll is recreated from message flag after overcast click, but the flags are not updated, thats why i have added the:

      this.updateMessageFlags();

into all overcast and overcastreset messages.

Reply:
Do you know of a case where a homebrew duration would fail?

Additionally
```js
 if (duration && item.duration.value.toLowerCase().includes(game.i18n.localize("Minutes")))
 ``` 
Why were the units of time capitalized? I'm not sure how it could be working because lowercase duration can never match "Minutes"

Lastly, 
If the `multiplier` variable is not being used it should be removed.

Reply:
Removed, also thats a funny thing - in en.json and other translations, Minutes, Rounds etc are all translated to lower case. IT was a small bummer that i found as well.